src/bin/huak/commands/mod.rs

```diff
@@ -17,36 +18,121 @@ pub(crate) mod remove;
 pub(crate) mod run;
 pub(crate) mod test;
 pub(crate) mod update;
-pub(crate) mod utils;
 pub(crate) mod version;
 
-pub fn args() -> Command<'static> {
-    let mut app = Command::new("huak");
-
-    let subcommands = vec![
-        activate::cmd(),
-        add::cmd(),
-        build::cmd(),
-        clean::cmd(),
-        clean_pycache::cmd(),
-        doc::cmd(),
-        help::cmd(),
-        fmt::cmd(),
-        init::cmd(),
-        install::cmd(),
-        lint::cmd(),
-        new::cmd(),
-        publish::cmd(),
-        remove::cmd(),
-        run::cmd(),
-        test::cmd(),
-        update::cmd(),
-        version::cmd(),
-    ];
-
-    for cmd in subcommands {
-        app = app.subcommand(cmd)
-    }
+// Main CLI struct.
+
+/// A Python package manager written in Rust inspired by Cargo.
+#[derive(Parser)]
+#[command(
+    version,
+    author,
+    about,
+    arg_required_else_help = true,
+    disable_help_subcommand = true
+)]
+pub struct Cli {
+    #[command(subcommand)]
+    pub command: Commands,
+}
+
+// List of commands.
+#[derive(Subcommand)]
+pub enum Commands {
+    /// Activate the project's virtual environment.
+    Activate,
+
+    /// Add a python module to the existing project.
+    Add {
+        dependency: String,
+        /// Adds an optional dependency.
+        #[arg(long)]
+        dev: bool,
+    },
+
+    /// Build tarball and wheel for the project.
+    Build,
+
+    /// Remove tarball and wheel from the built project.
+    Clean,
+
+    /// Remove all .pyc files and __pycache__ directores.
+    #[command(name = "clean-pycache")]
+    Cleanpycache,
+
+    /// Builds and uploads current project to a registry.
+    Doc {
+        /// Check if Python code is formatted.
+        #[arg(long)]
+        check: bool,
+    },
+
+    /// Display Huak commands and general usage information.
+    Help,
+
+    /// Format Python code.
+    Fmt {
+        /// Check if Python code is formatted.
+        #[arg(long)]
+        check: bool,
+    },
+
+    /// Initialize the existing project.
+    Init,
 
-    app
+    /// Install the dependencies of an existing project.
+    Install,
+
+    /// Lint Python code.
+    Lint,
+
+    /// Create a project from scratch.
+    New { path: Option<String> },
+
+    /// Builds and uploads current project to a registry.
+    Publish,
+
+    /// Remove a dependency from the project.
+    Remove { dependency: String },
+
+    /// Run a command within the project's environment context.
+    Run { command: String },
+
+    /// Test Python Code.
+    Test,
+
+    /// Update dependencies added to the project.
+    Update {
+        #[arg(default_value = "*")]
+        dependency: String,
+    },
+
+    /// Display the version of the project.
+    Version,
+}
```

Comment:
```suggestion
pub enum Commands {
    /// Activate the project's virtual environment.
    Activate,
    /// Add a python module to the existing project.
    Add {
        dependency: String,
        /// Adds an optional dependency.
        #[arg(long)]
        dev: bool,
    },
    /// Build tarball and wheel for the project.
    Build,
    /// Remove tarball and wheel from the built project.
    Clean,
    /// Remove all .pyc files and __pycache__ directores.
    #[command(name = "clean-pycache")]
    Cleanpycache,
    /// Builds and uploads current project to a registry.
    Doc {
        /// Check if Python code is formatted.
        #[arg(long)]
        check: bool,
    },
    /// Display Huak commands and general usage information.
    Help,
    /// Format Python code.
    Fmt {
        /// Check if Python code is formatted.
        #[arg(long)]
        check: bool,
    },
    /// Initialize the existing project.
    Init,
    /// Install the dependencies of an existing project.
    Install,
    /// Lint Python code.
    Lint,
    /// Create a project from scratch.
    New { path: Option<String> },
    /// Builds and uploads current project to a registry.
    Publish,
    /// Remove a dependency from the project.
    Remove { dependency: String },
    /// Run a command within the project's environment context.
    Run { command: String },
    /// Test Python Code.
    Test,
    /// Update dependencies added to the project.
    Update {
        #[arg(default_value = "*")]
        dependency: String,
    },
    /// Display the version of the project.
    Version,
}
```

Love it! Could we remove the white-space please?

Reply:
Already done!